### DISCUSSION TITLE
Does Amperfy support ALAC/FLAC on iOS?

### DISCUSSION THREAD

**Discussion by radiosilence** [BLeeEZ/amperfy#346] [Q&A]
My navidrome server has a decent amount of ALAC from Bandcamp and various other FLAC albums. Does it support them natively through iOS's audio drivers, or will navidrome have to transcode? I've selected RAW/Original in settings for both streamed and cached music.

**Comment by BLeeEZ** [comment]
The used audio engine doesn’t support FLAC. Server side transcoding is required for those files.

**Reply by radiosilence** [reply]
Thanks for your prompt response! Which engine is used? Something based on Core Audio or something custom?

**Reply by BLeeEZ** [reply]
AVPlayer: https://developer.apple.com/documentation/avfoundation/avplayer

**Reply by tehabe** [reply]
This answer confuses me, since I tried FLAC and ALAC on my Navidrome server and both formats work with Amperfy without a problem and without transcoding.

I still keep my music in FLAC and ALAC, god alone knows why.

**Reply by rossbates** [reply]
Same experience here. I had to set the format to Raw/Original in settings but I am streaming FLAC with no transcoder on the server. Thrilled it works but now curious why if AVplayer doesn't support it.

**Reply by soppelmann** [reply]
something is funky, transcoding fails but raw/original works. It even displays FLAC as the current format. :-)

**Reply by BLeeEZ** [reply]
In the meantime I replaced the audio engine. FLAC is supported now.
